### PR TITLE
plugin: Add basic unit tests for resourceTransitioner

### PR DESCRIPTION
**Pull request by sharnoff**
Found myself wanting these as part of #517, and it's been on the todo list for a while from #764.

Some of the tests currently aren't working as expected. They are configured to pass, acting as if the undesired outcome is expected. The bugfix will be in a separate PR.

---

**Notes for review:** There's still other methods I'd like to get test coverage for, so I think this doesn't completely fulfill #764. But, IMO it's a good start, at least -- and should definitely be useful for #517 :)

### CODE REVIEW

File: pkg/plugin/trans_test.go [hunk 1]
```diff
@@ -0,0 +1,521 @@
...
+			nodeReservedBefore: 16,
+			nodeBufferBefore:   0,
+
+			overbudget:        true,
+			verdict:           "node reserved 16 + 8 -> 24 of total 12",
+			nodeReservedAfter: 24,
+			nodeBufferAfter:   0,
+		},
+		{
+			name: "AlreadyOverbudgetWithBuffer",
```
Comment (petuhovskiy) [inline pkg/plugin/trans_test.go:113]: This config is the same as `AlreadyOverbudget`, I think you forgot to add buffer here

**Comment by petuhovskiy** [comment]
https://github.com/neondatabase/autoscaling/blob/2bf8d3e9e703e902f5963252d46da222528fa1ef/pkg/plugin/trans.go#L275

FYI this condition is always true, looks like a bug probably?

File: pkg/plugin/trans_test.go [hunk 2]
```diff
@@ -0,0 +1,521 @@
...
+				buffer:   0,
+				pressure: 0,
+			},
+			nodeAfter: node{
+				reserved: 13,
+				buffer:   0,
+				pressure: 0,
+			},
+		},
+	}
```
Comment (petuhovskiy) [inline pkg/plugin/trans_test.go:485]: Probably related to the FIXMEs above, found something in the `trans.go`:

First we set `forceApprovalMinimum` to:
https://github.com/neondatabase/autoscaling/blob/2bf8d3e9e703e902f5963252d46da222528fa1ef/pkg/plugin/trans.go#L191-L194

Then we use it as the minimum increase here:
https://github.com/neondatabase/autoscaling/blob/2bf8d3e9e703e902f5963252d46da222528fa1ef/pkg/plugin/trans.go#L315-L316

Currently it means that that plugin always allows scaling up to 2 times from the `lastPermit` value, probably a bug?

**Review by petuhovskiy — APPROVED** [review]

**Comment by sharnoff** [comment]
> FYI this condition is always true, looks like a bug probably?

Ah good catch, that is a bug, yeah. It should be `oldState.pod.Reserved`.

File: pkg/plugin/trans_test.go (hunk 2 above)
Comment (sharnoff) [reply pkg/plugin/trans_test.go:485]: Exactly, yup (previously opened a fix PR in #1025).

File: pkg/plugin/trans_test.go (hunk 1 above)
Comment (sharnoff) [reply pkg/plugin/trans_test.go:113]: Yup, good catch - thanks. Fixed in 2e8dbc2eff31dcdb09c13f77ab8b0cfd7ff22e2f

**Pull request merged**